library/src/test/java/com/novoda/downloadmanager/LiteBatchFileBuilderTest.java

```diff
@@ -0,0 +1,99 @@
+package com.novoda.downloadmanager;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class LiteBatchFileBuilderTest {
```

Comment:
Wondering if we should have a test / tests with a ludicrous number of Separators spaced throughout the `root` and the `paths`? At the moment the most extreme case we have is `/path/`.

Reply:
So, I agree we could do that (sanitise all elements throughout) but I think that could also be the user's job. I think accommodating for leading and trailing separators is enough (for now) to fix some weird edge cases.

Reply:
Hmmm, I have slightly changed `doesNotAddDuplicateSeparators` in bd9a16a824a8926f8bb08bb9521220996417fe05 to test the case where every single path element has leading and trailing separators.

Reply:
Sorry, my comment was hinting at what happens if you have `//path//` for example.